object/host_system.go

```diff
@@ -32,12 +32,31 @@ type HostSystem struct {
 	InventoryPath string
 }
 
+func (h HostSystem) String() string {
```

Comment:
I rather not have helpers that look like plain getters but are actually doing network calls under the hood. This makes it possible for your application to stall without you having any control over it (as is the case when you use the `context.Context` struct where you can explicitly cancel those network calls if they take too long).

Since they are calling out to the `Name` function anyway, could you remove them (we can chat about this in the PR)?